sycl/plugins/esimd_emulator/pi_esimd_emulator.cpp

```diff
@@ -900,6 +864,7 @@ pi_result piContextRelease(pi_context Context) {
   }
 
   if (--(Context->RefCount) == 0) {
+    std::lock_guard<std::mutex> Lock(Context->Addr2CmBufferSVMLock);
```

Comment:
For future: would be good to see comments why certain locks are global, and others are per-Context